frontend/src/modules/summary/productList/ProductListContainer.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component, ChangeEvent, ReactNode } from 'react';
+import { Card } from '@material-ui/core';
+
+import ProductList from './ProductList';
+import Product, { getVersionFromString } from 'models/Product';
+import Version from 'models/Version';
+
+interface Props {
+  list: Product[];
+  title: string;
+  handleFavouriteChange(product: Product): void;
+  handlePinChange(product: Product, version: Version, build: string, isPinned: boolean): void;
+}
+
+interface State {
+  searchContent: string;
+  expandedProductsList: string[];
+  selectedVersionMap: Record<string, Version | undefined>;
+}
+
+class ProductListContainer extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    const { list } = this.props;
+    const selectedVersionMap: Record<string, Version> = {};
+    for (const { name, versionList } of list) {
+      selectedVersionMap[name] = versionList[0];
+    }
+
+    this.state = {
+      searchContent: '',
+      expandedProductsList: [],
+      selectedVersionMap,
+    };
+  }
+
+  handleSearchProduct = (event: ChangeEvent<HTMLInputElement>): void => {
+    this.setState({ searchContent: event.target.value });
+  };
+
+  handleVersionSelected = (event: ChangeEvent<{ value: string }>, product: Product): void => {
+    this.setState(prevState => ({
+      ...prevState,
+      selectedVersionMap: { ...prevState.selectedVersionMap, [product.name]: getVersionFromString(product, event.target.value) },
```

Comment:
Should we look at using immer for this? Seems like a good candidate for it.
https://immerjs.github.io/immer/docs/example-setstate

Reply:
Yeah could be a good idea, I don't think we ever want this in redux so this will be a good point we can put it in here. I will look into it.